app/models/katello/repository.rb

```diff
@@ -860,20 +860,45 @@ def self.search_by_content_label(_key, operator, value)
     end
 
     def self.safe_render_container_name(repository, pattern = nil)
-      if (pattern && !pattern.blank?) || (repository.environment && !repository.environment.registry_name_pattern.empty?)
+      #            pattern provided / env pattern provided
+      #                 |  00 |  01 |  11 |  10
+      # ----------------+-----+-----+-----+------
+      # env exists /  00|   4 | n/a | n/a |   4
+      # is cv default 01|   2 | n/a | n/a |   4
+      #               11|   2 |   1 |   1 |   1
+      #               10|   3 |   1 |   1 |   1
+      #
+      # This table shows the name to render given the properties of
+      # the container provided. Branches numbered as ordered below.
+      #
+      # 1 - Render promotion pattern (or env pattern if no promo pattern)
```

Comment:
By  promotion pattern, do you mean the variable `pattern` in this method?

Reply:
I do. This pattern is set based on the registry name pattern of the lifecycle environment to which the CVV is being added. I believe it is blank and the env pattern takes over in some other cases. Is this still good to ship?

Reply:
All good!